extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1482,12 +1592,82 @@ export class Project implements ISqlProject {
 		const firstPropertyElement = propertyElements[0];
 		if (firstPropertyElement.childNodes.length !== 1) {
 			// Property items are expected to have simple string content
-			throw new Error(constants.invalidProjectPropertyValue(propertyName));
+			throw new Error(constants.invalidProjectPropertyValueInSqlProj(propertyName));
 		}
 
 		return firstPropertyElement.childNodes[0].nodeValue!;
 	}
 
+	/**
+	 * Sets the value of project property, overrides value if already present in project file.
+	 * @param propertyName Name of property
+	 * @param propertyValue Value of property
+	 */
+	private async setProjectPropertyValue(propertyName: string, propertyValue: string): Promise<void> {
+		if (this.projFileXmlDoc === undefined) {
+			return;
+		}
+
+		const propertyGroups = this.projFileXmlDoc.getElementsByTagName(constants.PropertyGroup);
+		let propertyGroup = propertyGroups.length > 0 ? propertyGroups[0] : null;
+		if (propertyGroup === null) {
+			propertyGroup = this.projFileXmlDoc.createElement(constants.PropertyGroup);
+			this.projFileXmlDoc.documentElement?.appendChild(propertyGroup);
+		}
+
+		const propertyElements = propertyGroup.getElementsByTagName(propertyName);
+		let propertyElement: Element | undefined;
+		if (propertyElements.length === 0) {
+			propertyElement = this.addProjectPropertyTag(propertyName);
+			propertyElement?.appendChild(this.projFileXmlDoc.createTextNode(propertyValue));
+		} else {
+			propertyElement = propertyElements[0];
+			if (propertyElement.childNodes.length > 0) {
+				propertyElement.replaceChild(this.projFileXmlDoc.createTextNode(propertyValue), propertyElement.childNodes[0]);
+			}
+		}
+
+		await this.serializeToProjFile(this.projFileXmlDoc);
+	}
+
+	/**
+	 * Adds empty project property tag
+	 * @param propertyTag Tag to add
+	 * @returns Added HTMLElement tag
+	 */
+	private addProjectPropertyTag(propertyTag: string): HTMLElement | undefined {
+		if (this.projFileXmlDoc === undefined) {
+			return;
+		}
+
+		const propertyGroups = this.projFileXmlDoc.getElementsByTagName(constants.PropertyGroup);
+		let propertyGroup = propertyGroups.length > 0 ? propertyGroups[0] : null;
+		if (propertyGroup === null) {
+			propertyGroup = this.projFileXmlDoc.createElement(constants.PropertyGroup);
+			this.projFileXmlDoc.documentElement?.appendChild(propertyGroup);
+		}
+
+		const propertyElement = this.projFileXmlDoc.createElement(propertyTag);
+		propertyGroup.appendChild(propertyElement);
+		return propertyElement;
+	}
+
+	/**
+	 * Remove first occurrence of project property
+	 * @param propertyTag Tag to remove
+	 */
+	private removeProjectPropertyTag(propertyTag: string) {
+		const propertyGroups = this.projFileXmlDoc!.getElementsByTagName(constants.PropertyGroup);
```

Comment:
I think we should null-check `this.projFileXmlDoc` here as well, instead of claiming it's not empty - again, if project is not loaded, then there is nothing to remove from.